src/js/common/schemaform/helpers.js

```diff
@@ -445,25 +449,25 @@ export function removeHiddenData(schema, data) {
       if (typeof data[next] !== 'undefined') {
         const nextData = removeHiddenData(schema.properties[next], data[next]);
 
+        // if the data was removed, then just unset it
         if (typeof nextData === 'undefined') {
           return _.unset(next, current);
         }
+
+        // if data was updated (like a nested prop was removed), update it
+        if (nextData !== data[next]) {
```

Comment:
In the case when `updatedSchema.type === 'object'. I think it's because the recursive call, along with the path argument, ensures that the traversal makes it down to the leaves of any nested properties, so for objects we can just check whether `nextData === 'undefined'` and un-set it if so. LMK if that's accurate.